nRF9160Master/src/Temp_MCP/TempAdcHandler.c

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file    Temp_Adchandler.c
+ * @author : Devendu
+ * @brief   File to calculate the temperature using MCP9700
+ * @date    09-04-2023
+ */
+/***********************************INCLUDES**************************/
+#include <nrfx_example.h>
+#include <saadc_examples_common.h>
+#include <nrfx_saadc.h>
+#include <zephyr/kernel.h>
+#include "TempAdcHandler.h"
+
+
+/************************************MACROS***************************/
+#define NRFX_LOG_MODULE                 EXAMPLE
+#define NRFX_EXAMPLE_CONFIG_LOG_ENABLED 1
+#define NRFX_EXAMPLE_CONFIG_LOG_LEVEL   3
+#include <nrfx_log.h>
+
+/** @brief Symbol specifying analog input to be observed by SAADC channel 0. */
+#define CH0_AIN ANALOG_INPUT_TO_SAADC_AIN(ANALOG_INPUT_A0)
+
+// Temperature coefficient in mV/°C
+#define TC_MV_PER_C 10.0
+
+// Output voltage at 0°C in mV
+#define V0_C_MV 440
+
+
+/************************************GLOBALS**************************/
+
+/** @brief Array specifying GPIO pins used to test the functionality of SAADC. */
+static uint8_t m_out_pins[1] = {LOOPBACK_PIN_1B, LOOPBACK_PIN_2B, LOOPBACK_PIN_3B};
+
+/** @brief Samples buffer defined with the size of @ref CHANNEL_COUNT symbol to store values from each channel ( @ref m_multiple_channels). */
+static nrf_saadc_value_t samples_buffer[1];
+
+/** @brief SAADC channel configuration structure for single channel use. */
+static const nrfx_saadc_channel_t m_single_channel = NRFX_SAADC_DEFAULT_CHANNEL_SE(CH0_AIN, 0);
```

Comment:
Is there any specific reason make this variable global. ? It is only used in _initialize_saadc()_ function

Reply:
it is removed as global variable and declared inside the   initialize_saadc() function